### DISCUSSION TITLE
Stimulus controllers not registers

### DISCUSSION THREAD

**Discussion by serhiihiba** [ElMassimo/stimulus-vite-helpers#10] [Q&A]
```js
import { Application } from '@hotwired/stimulus'
import { registerControllers } from 'stimulus-vite-helpers'

const application = Application.start()
application.debug = false
const controllers = import.meta.globEager('../../../../components/ui/bentries/log/**/*_controller.js')
console.log("import.meta.globEager =>", controllers);
window.Stimulus   = application
registerControllers(application, controllers)
```

<img width="1208" alt="image" src="https://github.com/ElMassimo/stimulus-vite-helpers/assets/31795690/75799119-84f6-492a-9864-5a8746a5b560">

As you can see, `window.Stimulus.controllers` is empty, is that a bug or just my issue?

That works by the old way:

```js
import { Application } from '@hotwired/stimulus'

const application = Application.start()
application.debug = false
window.Stimulus   = application
import LogComponent from "../../../../components/ui/bentries/log/log_component/log_component_controller";
application.register("ui--bentries--log--log-component", LogComponent);
```

<img width="1017" alt="image" src="https://github.com/ElMassimo/stimulus-vite-helpers/assets/31795690/f5a8e671-411e-4784-9a8c-43a9a736c56f">

**Comment by ElMassimo** [comment]
Hi Serhii!

Try placing the `globEager` in a file at the root of the controllers, so that `../../../../` does not become part of the glob keys:

```js
export const controllers = import.meta.globEager('./components/ui/bentries/log/**/*_controller.js')
```

**Comment by serhiihiba** [comment]
@ElMassimo thx for the fast response! But that don't work for me
<img width="235" alt="image" src="https://github.com/ElMassimo/stimulus-vite-helpers/assets/31795690/3de8c23f-5797-4e72-bc40-2861504aa7a2">

```json
# config/vite.json
{
  "all": {
    "sourceCodeDir": "app",
    "watchAdditionalPaths": []
  },
  "development": {
    "autoBuild": true,
    "publicOutputDir": "vite-dev",
    "port": 3036
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite-test",
    "port": 3037
  }
}
```

```TS
//vite.config.ts
import { defineConfig } from 'vite'
import RubyPlugin from 'vite-plugin-ruby'

export default defineConfig({
  plugins: [
    RubyPlugin(),
  ],
})
```

```json
#package.json
{
  "name": "app",
  "private": "true",
  "type": "module",
  "dependencies": {
    "@hotwired/stimulus": "^3.0.1",
    "@rails/actioncable": "^6.1.5",
    "@tailwindcss/aspect-ratio": "^0.4.2",
    "@tailwindcss/container-queries": "^0.1.1",
    "@tailwindcss/forms": "^0.5.4",
    "@tailwindcss/typography": "^0.5.9",
    "birdeljs": "^3.3.4",
    "esbuild": "^0.14.38",
    "stimulus-vite-helpers": "^3.0.1",
    "tailwindcss": "^3.3.3"
  },
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "serve": "vite preview"
  },
  "devDependencies": {
    "vite": "^4.3.0",
    "vite-plugin-ruby": "^3.2.0"
  }
}
```

```js
// app/javascript/ui/bentries/log/index.js

import { Application } from '@hotwired/stimulus'
import { registerControllers } from 'stimulus-vite-helpers'

const application = Application.start()
application.debug = false
const controllers = import.meta.globEager('./components/ui/bentries/log/**/*_controller.js')
console.log("import.meta.globEager =>", controllers);
window.Stimulus   = application
registerControllers(application, controllers)
```

But this way gives me a list of imported modules
```js
const controllers = import.meta.globEager('../../../../components/ui/bentries/log/**/*_controller.js')
```
<img width="1222" alt="image" src="https://github.com/ElMassimo/stimulus-vite-helpers/assets/31795690/5de1e1fb-cd41-4a49-b295-60db5ba96325">

**Comment by serhiihiba** [comment]
@ElMassimo maybe this can help somehow too
<img width="1427" alt="image" src="https://github.com/ElMassimo/stimulus-vite-helpers/assets/31795690/699a9570-28df-4ccf-ba19-5727ed256bc0">

**Comment by ElMassimo** [comment]
To clarify, I'm suggesting you have this:

```js
export const controllers = import.meta.globEager('./ui/bentries/log/**/*_controller.js')
```

in the root of your app (for example `app/components/index.js`), and then import from there:

```js
import { Application } from '@hotwired/stimulus'
import { registerControllers } from 'stimulus-vite-helpers'
import { controllers } from '../../../../components'

const application = Application.start()
registerControllers(application, controllers)
```

Based on the screenshot you provided:
![image](https://github.com/ElMassimo/stimulus-vite-helpers/assets/1158253/4ded26ae-98b0-4e89-ac52-af238ee68881)

you have not updated the path accordingly, or have not placed the file in the root.

Verify that logging `controllers` has the modules you would expect, or adjust the path accordingly.

**Comment by ElMassimo** [comment]
@serhiijun To minimize confusion, instead of using relative imports it's preferable to use _[Import Aliases](https://vite-ruby.netlify.app/guide/development.html#import-aliases-%F0%9F%91%89)_.

For example, you can define an alias to the `@components` directory by using [`resolve.alias`](https://vite-ruby.netlify.app/config/#watchadditionalpaths).
